pkg/config/config.go

```diff
@@ -81,14 +86,77 @@ func (c *Config) ConfigureRules() {
 }
 
 func loadConfig(filename string) (c Config, err error) {
+	if isValidURL(filename) {
+		// if fileName is a valid URL, we will download and set it to the config
+		log.Debug().Str("url", filename).Msg("Downloading file from")
+		// hardcoding this file and saving to root directory
+		downloadedFile := "downloadedRules.yaml"
+		err := DownloadFile(downloadedFile, filename)
+		if err != nil {
+			return c, err
+		}
+		filename = downloadedFile
+		log.Debug().Str("filename", filename).Msg("Saved remote config to local file.")
+	}
+
+	// TO DO = Need to add more error handling on reading/validating yamlFile file - issue # 16
 	yamlFile, err := ioutil.ReadFile(filename)
+	log.Debug().Str("filename", filename).Msg("Adding custom ruleset from")
 	if err != nil {
 		return c, err
 	}
-
 	return c, yaml.Unmarshal(yamlFile, &c)
 }
 
+// isValidUrl tests a string to determine if it is a valid URL or not
+func isValidURL(toTest string) bool {
+	_, err := url.ParseRequestURI(toTest)
+	if err != nil {
+		return false
+	}
+
+	u, err := url.Parse(toTest)
+	if err != nil || u.Scheme == "" || u.Host == "" {
+		return false
+	}
+	log.Debug().Str("remoteConfig", toTest).Msg("Valid URL for remote config.")
+	return true
+}
+
+// downloads file from url to set filepath
+func DownloadFile(filepath string, url string) error {
+	// Removing gosec lint warning - as we have to pass in user specified url for remote config
```

Comment:
This comment looks like it's from when we were doing `http.Get()` to document that we were suppressing the linting warnings -- is it still needed?

Reply:
Good call - forgot to remove the warning !